cmd/lambda/leases/get_test.go

```diff
@@ -1,66 +1,163 @@
-			ID:             "unique-id",
-			AccountID:      "123456789",
-			PrincipalID:    "test",
-			LeaseStatus:    db.Active,
-			LastModifiedOn: 1561149393,
+
+		cfgBuilder := &config.ConfigurationBuilder{}
+		svcBuilder := &config.ServiceBuilder{Config: cfgBuilder}
+
+		leaseSvc := mocks.Servicer{}
+		leaseSvc.On("Get", *expectedLease.ID).Return(
+			expectedLease, nil,
+		)
+		svcBuilder.Config.WithService(&leaseSvc)
+		_, err := svcBuilder.Build()
+
+		assert.Nil(t, err)
+		if err == nil {
+			Services = svcBuilder
 		}
-		mockDb := mocks.DBer{}
-		mockDb.On("GetLeaseByID", "unique-id").Return(expectdLease, nil)
-		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/leases/unique-id"}
 
-		dao = &mockDb
+		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/leases/unique-id"}
 
 		actualResponse, err := Handler(context.TODO(), mockRequest)
 		require.Nil(t, err)
 
-		parsedResponse := &response.LeaseResponse{}
-		err = json.Unmarshal([]byte(actualResponse.Body), parsedResponse)
-		require.Nil(t, err)
-
-		require.Equal(t, expectedLeaseResponse, parsedResponse, "Returns a single lease.")
-		require.Equal(t, actualResponse.StatusCode, 200, "Returns a 200.")
+		require.Equal(t, MockAPIResponse(http.StatusOK, "{\"accountId\":\"123456789\",\"principalId\":\"test\",\"id\":\"unique-id\",\"leaseStatus\":\"Active\",\"lastModifiedOn\":1561149393}\n"), actualResponse, "Returns a single lease.")
 	})
 
-	t.Run("When the query fails", func(t *testing.T) {
-		expectedError := errors.New("Error")
-		mockDb := mocks.DBer{}
-		mockDb.On("GetLeaseByID", "unique-id").Return(nil, expectedError)
-		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/leases/unique-id"}
+	t.Run("When the handler invoking get and get fails", func(t *testing.T) {
+		expectedLease := &lease.Lease{
+			ID: ptrString("unique-id"),
+		}
+
+		expectedError := gErrors.New("error")
+		cfgBuilder := &config.ConfigurationBuilder{}
+		svcBuilder := &config.ServiceBuilder{Config: cfgBuilder}
 
-		dao = &mockDb
+		leaseSvc := mocks.Servicer{}
+		leaseSvc.On("Get", *expectedLease.ID).Return(
+			expectedLease, expectedError,
+		)
+		svcBuilder.Config.WithService(&leaseSvc)
+		_, err := svcBuilder.Build()
+
+		assert.Nil(t, err)
+		if err == nil {
+			Services = svcBuilder
+		}
+
+		mockRequest := events.APIGatewayProxyRequest{HTTPMethod: http.MethodGet, Path: "/leases/unique-id"}
 
 		actualResponse, err := Handler(context.TODO(), mockRequest)
 		require.Nil(t, err)
 
 		require.Equal(t, actualResponse.StatusCode, 500, "Returns a 500.")
-		require.Equal(t, actualResponse.Body, "{\"error\":{\"code\":\"ServerError\",\"message\":\"Failed Get on Lease unique-id\"}}", "Returns an error response.")
+		require.Equal(t, MockAPIErrorResponse(http.StatusInternalServerError, "{\"error\":{\"message\":\"unknown error\",\"code\":\"ServerError\"}}\n"), actualResponse, "Returns an error response.")
```

Comment:
I would prefer what is in the accounts controller here.
```
assert.Equal(t, tt.expResp.StatusCode, resp.StatusCode)
assert.Equal(t, tt.expResp.Body, string(body))
```
or even better would probably be using https://godoc.org/github.com/stretchr/testify/assert#JSONEq

Reply:
fixed